src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -77,37 +77,49 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
       $query->condition($bundle_key, $bundles, 'IN');
     }
 
-    $user_groups = $this->getUserGroups();
-    if (!$user_groups) {
+    $user = User::load(\Drupal::currentUser()->id());
+
+    if (!$user) {
+      // @todo: What to do with anonymous user?
```

Comment:
Below we do:

```
// User doesn't have permission to select any group so falsify this
// query.
$query->condition($identifier_key, -1);
```

for authenticated users. It feels fitting to do the same for anonymous.
Other access checks apply so in most use cases anonymous won't be able to reference groups, but I think one has to be defensive.